src/openforms/authentication/contrib/digid/tests/test_signicat_integration.py

```diff
@@ -0,0 +1,608 @@
+
+        our_faux_redirect = self.client.get(f"{login_url}?next={form_url}", follow=True)
+        # do the JS submit to get redirected to signicat broker
+        method, redirect_url, form_values = _parse_form(our_faux_redirect)
+        session.request(method, redirect_url, data=form_values)
+        select_digid_sim = SIGNICAT_BROKER_BASE / "authn/simulator/selection/digid"
+        sim_response = session.get(select_digid_sim)
+        sim_method, sim_action_url, sim_form = _parse_form(sim_response)
+        auth_response = session.request(
+            sim_method, sim_action_url, data=sim_form, allow_redirects=False
+        )
+        # not redirected
+        self.assertFalse(auth_response.ok)
+
+
+# class SignicatWithEncryptedAssertionsTests(SignicatDigiDIntegrationTests):
+#     """This doens't work; after changing our DigidConfiguration, we need to send our updated
+#     SAML metadata to the broker (Signicat)
+#
+#     To test this we'd need to configure a separate entityID
+#     (perhaps with a different certificate/key pair) and set it up with Signicat
+#     """
+#
+#
+#     @classmethod
+#     def setUpTestData(cls):
+#         super().setUpTestData()
+#
+#         config = DigidConfiguration.get_solo()
+#         config.want_assertions_encrypted = True
+#         config.save()
+
+# class SignicatWithUnsignedAssertionsTests(SignicatDigiDIntegrationTests):
+#     """These pass, but Signicat signs assertions, because it will follow
+#     the metadata we uploaded previously, which wanted assertions signed.
+#
+#     Our SAML metadata discovery url is unreachable...
+#     https://localhost:8000/discovery/digid-eherkenning/digid
+#
+#     I don't know if this is used in a live, non-testing environment.
+#     """
+#
+#
+#     @classmethod
+#     def setUpTestData(cls):
+#         super().setUpTestData()
+#
+#         config = DigidConfiguration.get_solo()
+#         config.want_assertions_signed = False
+#         config.save()
+
+# Same same, but different
+# class SignicatWithSingnedAndEncryptedAssertionsTests(SignicatDigiDIntegrationTests):
+#     @classmethod
+#     def setUpTestData(cls):
+#         super().setUpTestData()
+#
+#         config = DigidConfiguration.get_solo()
+#         config.want_assertions_signed = True
+#         config.want_assertions_encrypted = True
+#         config.save()
+
+
+# Helper functions
+
+# poor person's enum.StrEnum
+Method = Union[Literal["get"], Literal["post"]]
+Response = Union[TemplateResponse, requests.Response]
+
+
+def _parse_value(el):
+    match el.name:
+        case "input":
+            return el.get("value", "")
+        case "select":
+            selected = el.find("option", selected=True)
+            return selected.get("value", "")
+
+
+def _parse_form(response: Response) -> tuple[Method, str, dict[str, str]]:
+    "Extract method, action URL and form values from html content"
+    # doesn't handle radio/checkbox... Need more? Look for a library don't flesh this out.
```

Comment:
I replace this entire thing with an existing library (and dependency), such as WebTest ;)